### DISCUSSION TITLE
Changing circuit breaker state in case of certain condition/ computation result

### DISCUSSION THREAD

**Discussion by rakeshvende** [resilience4j/resilience4j#1933] [General]
Hello there, 

For one particular scenario, need to open the circuit breaker in case of high computation load before it actually breaks the system.  This computational threshold can be breached in single call or across multiple calls, if monitored for certain period.  

This is not a error scenario where call is failing, its from computation resource consumption perspective. With current design of resilience4j circuit breaker, can we handle this particular use case?

**Comment by RobWin** [comment]
Hi,
you can always change the state of circuitbreaker manually.
See: 
https://github.com/resilience4j/resilience4j/blob/master/resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreaker.java#L527-L535
